### PR TITLE
Modernize file upload

### PR DESCRIPTION
**Pull request by MikhailoKhraban**
- Replace form-multipart upload with chunked (v3) upload
- Support resumable upload for files larger than 5MB

### CODE REVIEW

File: lib/Management/FileUploader.php [hunk 1]
```diff
@@ -64,30 +75,22 @@ public function getUploadConfiguration(UploadConfigurationRequest $uploadConfigu
...
 
-        $options['multipart'] = $form;
-
-        $restResponse = $this->authenticatedHTTPClient->post(
-	        $uploadConfigurationResponse->getUploadUrl(),
-	        array(),
-	        $options
-        );
+        $size = filesize($source);
+        $options['lifecycle'] = ['age' => 30, 'action' => 'delete'];
```
Comment (Elad-Laufer) [inline lib/Management/FileUploader.php:84]: Remove this, any file file uploaded will be deleted after 30 minutes.

File: lib/Management/FileUploader.php [hunk 2]
```diff
@@ -64,30 +75,22 @@ public function getUploadConfiguration(UploadConfigurationRequest $uploadConfigu
 	 * @return array
 	 */
     public function uploadFile($path, $mimeType, $source, $acl = null, $options = []) {
```
Comment (Elad-Laufer) [inline lib/Management/FileUploader.php:77]: The default should be explicit, and I recommend to default to 'private'

File: lib/Tus/TusClient.php [hunk 3]
```diff
@@ -0,0 +1,98 @@
+<?php
+namespace Wix\Mediaplatform\Tus;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\GuzzleException;
+use GuzzleHttp\Psr7\Request;
+use Wix\Mediaplatform\Model\Response\RestResponse;
+
+
+class TusClient
```
Comment (Elad-Laufer) [inline lib/Tus/TusClient.php:10]: Why not encapsulate https://github.com/ankitpokhrel/tus-php ?

**Review by Elad-Laufer — CHANGES_REQUESTED** [review]

File: lib/Tus/TusClient.php [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        $response = $this->httpClient->send($request, $options);
+        if ($response->getStatusCode() == 201) {
+            $headers = $response->getHeaders();
+            return  $headers['Location'][0];
+        } else {
+            new \Exception('TusClient can\t get location');
+        }
+    }
+
+    protected function finilize($url, $options) {
```
Comment (lelikg) [inline lib/Tus/TusClient.php:46]:
```suggestion
    protected function finalize($url, $options) {
```

File: lib/Tus/TusClient.php [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+                if ($retry >= self::RETRY_COUNT) {
+                    throw $e;
+                }
+            }
+        }
+
+        return $this->finilize(self::BASE_ENDPOINT . '/' . $token, $options);
+    }
+}
+?>
```
Comment (lelikg) [inline lib/Tus/TusClient.php:98]: please don't leave closing tags in php files

**Review by lelikg — CHANGES_REQUESTED** [review]
1. there are no tests
2. some other minor changes

**Review by lelikg — APPROVED** [review]

**Pull request merged**